src/test/java/com/digitalpebble/spruce/modules/boavizta/BoaviztAPITest.java

```diff
@@ -2,26 +2,267 @@
 
 package com.digitalpebble.spruce.modules.boavizta;
 
+import com.digitalpebble.spruce.CURColumn;
+import com.digitalpebble.spruce.Column;
+import com.digitalpebble.spruce.SpruceColumn;
 import com.digitalpebble.spruce.Utils;
 import org.apache.spark.sql.Row;
 import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
 import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 public class BoaviztAPITest extends AbstractBoaviztaTest {
 
-    BoaviztAPI api = new BoaviztAPI();
-    StructType schema = Utils.getSchema(api);
+    private BoaviztAPI api;
+    private StructType schema;
+
+    @BeforeEach
+    void setUp() {
+        api = new BoaviztAPI();
+        schema = Utils.getSchema(api);
+    }
+
+    @Test
+    void testColumnsNeeded() {
+        Column[] needed = api.columnsNeeded();
+        assertEquals(4, needed.length);
+        assertEquals(CURColumn.PRODUCT_INSTANCE_TYPE, needed[0]);
+        assertEquals(CURColumn.PRODUCT_SERVICE_CODE, needed[1]);
+        assertEquals(CURColumn.LINE_ITEM_OPERATION, needed[2]);
+        assertEquals(CURColumn.LINE_ITEM_PRODUCT_CODE, needed[3]);
+    }
+
+    @Test
+    void testColumnsAdded() {
+        Column[] added = api.columnsAdded();
+        assertEquals(1, added.length);
+        assertEquals(SpruceColumn.ENERGY_USED, added[0]);
+    }
 
     @Test
-    void testGetEC2() {
+    void testInitWithDefaultAddress() {
+        Map<String, Object> params = new HashMap<>();
+        api.init(params);
+        // Should use default address "http://localhost:5000"
+        assertNotNull(api);
+    }
+
+    @Test
+    void testInitWithCustomAddress() {
+        Map<String, Object> params = new HashMap<>();
+        params.put("address", "http://custom-host:8080");
+        api.init(params);
+        // The address should be set to the custom value
+        assertNotNull(api);
+    }
+
+    @Test
+    void testProcessWithNullInstanceType() {
         Object[] values = new Object[]{"AWSDataTransfer", null, null, null, null};
         Row row = new GenericRowWithSchema(values, schema);
         Row enriched = api.process(row);
-        // missing values comes back as it was
+        
+        // Should return the original row unchanged
+        assertEquals(row, enriched);
+    }
+
+    @Test
+    void testProcessWithNullOperation() {
```

Comment:
given that these tests have a similar behaviour and check that the inputs are not changed, they differ only in their inputs. What about having a parameterized test instead?
https://www.baeldung.com/parameterized-tests-junit-5#6-method